### PR TITLE
Issue #1733 - implement TOKEN :above and :below modifiers

### PR DESCRIPTION
**Pull request by michaelwschroeder**
Signed-off-by: Mike Schroeder <mschroed@us.ibm.com>

### CODE REVIEW

**Review by tbieste — APPROVED** [review]
LGTM

**Comment by JohnTimm** [comment]
Since the method `populateValueSetCodesSubSegment` now handles both types of modifiers the name of the method should probably change to reflect that instead of mentioning "ValueSet". Maybe just `populateCodesSubSegment`?

**Comment by michaelwschroeder** [comment]
> 
> 
> Since the method `populateValueSetCodesSubSegment` now handles both types of modifiers the name of the method should probably change to reflect that instead of mentioning "ValueSet". Maybe just `populateCodesSubSegment`?

Done.

File: docs/src/pages/Conformance.md [hunk 1]
```diff
@@ -172,7 +172,10 @@ FHIR search modifiers are described at https://www.hl7.org/fhir/R4/search.html#m
 
 Due to performance implications, the `:exact` modifier should be used for String search parameters when possible.
 
-The `:above`, and `:below` modifiers for Token search parameters are not yet supported by the IBM FHIR server. Use of these modifiers will result in an HTTP 400 error with an OperationOutcome that describes the failure.
+The `:above` and `:below` modifiers for Token search parameters are supported, with the following restrictions:
+* The search parameter value must have the form of `[system]|[code]`.
+* The referenced code system must exist in the FHIR registry.
+* The referenced code must exist in the code system.
```
Comment (lmsurpre) [inline docs/src/pages/Conformance.md:178]: this particular one doesn't seem like much of a restriction to me :-)

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/JDBCConstants.java [hunk 2]
```diff
@@ -137,7 +137,8 @@
         supportedModifiersMap.put(Type.STRING, Arrays.asList(Modifier.EXACT, Modifier.CONTAINS, Modifier.MISSING));
         supportedModifiersMap.put(Type.REFERENCE, Arrays.asList(Modifier.TYPE, Modifier.MISSING, Modifier.IDENTIFIER));
         supportedModifiersMap.put(Type.URI, Arrays.asList(Modifier.BELOW, Modifier.ABOVE, Modifier.MISSING));
-        supportedModifiersMap.put(Type.TOKEN, Arrays.asList(Modifier.MISSING, Modifier.NOT, Modifier.OF_TYPE, Modifier.IN, Modifier.NOT_IN, Modifier.TEXT));
+        supportedModifiersMap.put(Type.TOKEN, Arrays.asList(Modifier.MISSING, Modifier.NOT, Modifier.OF_TYPE,
+            Modifier.IN, Modifier.NOT_IN, Modifier.TEXT, Modifier.ABOVE, Modifier.BELOW));
```
Comment (lmsurpre) [inline fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/JDBCConstants.java:141]: style suggestion: double-indent wrapped lines
```suggestion
                Modifier.IN, Modifier.NOT_IN, Modifier.TEXT, Modifier.ABOVE, Modifier.BELOW));
```

**Review by lmsurpre — APPROVED** [review]
LGTM

File: docs/src/pages/Conformance.md (hunk 1 above)
Comment (michaelwschroeder) [reply docs/src/pages/Conformance.md]: Good point, not really a restriction - removed.

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/JDBCConstants.java (hunk 2 above)
Comment (michaelwschroeder) [reply fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/JDBCConstants.java]: Done.

**Pull request merged**